## **Scomposizione Titanic Dataset**

In [182]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

import warnings
warnings.filterwarnings("ignore")

class_names = {1: 'First Class', 2: 'Second Class', 3: 'Third Class'}

Import totale:

In [183]:
full = pd.read_csv("../Dataset/Titanic/train.csv")
# test = pd.read_csv("../Dataset/Titanic/test.csv")
# full = pd.concat([train, test])
full

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

#### **1) Mapping Nomi - Passenger ID:**

In [184]:
ids = full[['PassengerId', 'Name']]
ids.columns = ['Passenger Id', 'Name Orig']
ids['Surname'] = ids['Name Orig'].apply(lambda x: x.split(", ")[0])
ids['Name'] = ids['Name Orig'].apply(lambda x: x.split(", ")[1])
ids.sort_values(by='Passenger Id', inplace=True)
ids.reset_index(drop=True, inplace=True)
ids

Passenger Id                                          Name Orig  \
0               1                            Braund, Mr. Owen Harris   
1               2  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2               3                             Heikkinen, Miss. Laina   
3               4       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4               5                           Allen, Mr. William Henry   
..            ...                                                ...   
886           887                              Montvila, Rev. Juozas   
887           888                       Graham, Miss. Margaret Edith   
888           889           Johnston, Miss. Catherine Helen "Carrie"   
889           890                              Behr, Mr. Karl Howell   
890           891                                Dooley, Mr. Patrick   

       Surname                                        Name  
0       Braund                             Mr. Owen Harris  
1      Cumings  Mrs. John Bradley (Florence Briggs Thayer)  
2    Heikkinen                                 Miss. Laina  
3     Futrelle          Mrs. Jacques Heath (Lily May Peel)  
4        Allen                           Mr. William Henry  
..         ...                                         ...  
886   Montvila                                 Rev. Juozas  
887     Graham                        Miss. Margaret Edith  
888   Johnston              Miss. Catherine Helen "Carrie"  
889       Behr                             Mr. Karl Howell  
890     Dooley                                 Mr. Patrick  

[891 rows x 4 columns]

In [185]:
ids.drop(columns='Name Orig', inplace=True)
ids

Passenger Id    Surname                                        Name
0               1     Braund                             Mr. Owen Harris
1               2    Cumings  Mrs. John Bradley (Florence Briggs Thayer)
2               3  Heikkinen                                 Miss. Laina
3               4   Futrelle          Mrs. Jacques Heath (Lily May Peel)
4               5      Allen                           Mr. William Henry
..            ...        ...                                         ...
886           887   Montvila                                 Rev. Juozas
887           888     Graham                        Miss. Margaret Edith
888           889   Johnston              Miss. Catherine Helen "Carrie"
889           890       Behr                             Mr. Karl Howell
890           891     Dooley                                 Mr. Patrick

[891 rows x 3 columns]

In [186]:
ids.to_csv("../out/Titanic/passenger_id.csv", index=False)

#### **2) Anagrafe passeggeri**:

In [187]:
registry = full[['Name', 'Sex', 'Age', 'Pclass', 'Embarked']]
registry.sort_values(by='Name', inplace=True)
registry.rename(columns={'Name': 'Name Orig', 'Pclass': 'Pclass Orig'}, inplace=True)
registry.reset_index(drop=True, inplace=True)
registry

Name Orig     Sex   Age  Pclass Orig Embarked
0                      Abbing, Mr. Anthony    male  42.0            3        S
1              Abbott, Mr. Rossmore Edward    male  16.0            3        S
2         Abbott, Mrs. Stanton (Rosa Hunt)  female  35.0            3        S
3                      Abelson, Mr. Samuel    male  30.0            2        C
4    Abelson, Mrs. Samuel (Hannah Wizosky)  female  28.0            2        C
..                                     ...     ...   ...          ...      ...
886                de Mulder, Mr. Theodore    male  30.0            3        S
887              de Pelsmaeker, Mr. Alfons    male  16.0            3        S
888              del Carlo, Mr. Sebastiano    male  29.0            2        C
889        van Billiard, Mr. Austin Blyler    male  40.5            3        S
890            van Melkebeke, Mr. Philemon    male   NaN            3        S

[891 rows x 5 columns]

In [188]:
registry['Pclass'] = [class_names[registry['Pclass Orig'].iloc[i]] for i in range(len(registry.index))]
registry['Surname'] = registry['Name Orig'].apply(lambda x: x.split(", ")[0])
registry['Name'] = registry['Name Orig'].apply(lambda x: x.split(", ")[1])
registry['Sex'] = np.where(registry['Sex'] == 'male', 'Male', 'Female')
registry

Name Orig     Sex   Age  Pclass Orig  \
0                      Abbing, Mr. Anthony    Male  42.0            3   
1              Abbott, Mr. Rossmore Edward    Male  16.0            3   
2         Abbott, Mrs. Stanton (Rosa Hunt)  Female  35.0            3   
3                      Abelson, Mr. Samuel    Male  30.0            2   
4    Abelson, Mrs. Samuel (Hannah Wizosky)  Female  28.0            2   
..                                     ...     ...   ...          ...   
886                de Mulder, Mr. Theodore    Male  30.0            3   
887              de Pelsmaeker, Mr. Alfons    Male  16.0            3   
888              del Carlo, Mr. Sebastiano    Male  29.0            2   
889        van Billiard, Mr. Austin Blyler    Male  40.5            3   
890            van Melkebeke, Mr. Philemon    Male   NaN            3   

    Embarked        Pclass        Surname                          Name  
0          S   Third Class         Abbing                   Mr. Anthony  
1          S   Third Class         Abbott           Mr. Rossmore Edward  
2          S   Third Class         Abbott      Mrs. Stanton (Rosa Hunt)  
3          C  Second Class        Abelson                    Mr. Samuel  
4          C  Second Class        Abelson  Mrs. Samuel (Hannah Wizosky)  
..       ...           ...            ...                           ...  
886        S   Third Class      de Mulder                  Mr. Theodore  
887        S   Third Class  de Pelsmaeker                    Mr. Alfons  
888        C  Second Class      del Carlo                Mr. Sebastiano  
889        S   Third Class   van Billiard             Mr. Austin Blyler  
890        S   Third Class  van Melkebeke                  Mr. Philemon  

[891 rows x 8 columns]

In [189]:
registry_ok = pd.DataFrame({'Surname': registry['Surname'], 'Name': registry['Name'], 'Sex': registry['Sex'], 'Age': registry['Age'], 'Pclass': registry['Pclass'], 'Embarked': registry['Embarked']})
registry_ok

Surname                          Name     Sex   Age        Pclass  \
0           Abbing                   Mr. Anthony    Male  42.0   Third Class   
1           Abbott           Mr. Rossmore Edward    Male  16.0   Third Class   
2           Abbott      Mrs. Stanton (Rosa Hunt)  Female  35.0   Third Class   
3          Abelson                    Mr. Samuel    Male  30.0  Second Class   
4          Abelson  Mrs. Samuel (Hannah Wizosky)  Female  28.0  Second Class   
..             ...                           ...     ...   ...           ...   
886      de Mulder                  Mr. Theodore    Male  30.0   Third Class   
887  de Pelsmaeker                    Mr. Alfons    Male  16.0   Third Class   
888      del Carlo                Mr. Sebastiano    Male  29.0  Second Class   
889   van Billiard             Mr. Austin Blyler    Male  40.5   Third Class   
890  van Melkebeke                  Mr. Philemon    Male   NaN   Third Class   

    Embarked  
0          S  
1          S  
2          S  
3          C  
4          C  
..       ...  
886        S  
887        S  
888        C  
889        S  
890        S  

[891 rows x 6 columns]

In [190]:
registry_ok.to_csv("../out/Titanic/board_registry.csv", index=False)

#### **3) Biglietteria**

In [191]:
tickets = full[['Name', 'Ticket', 'Fare', 'Cabin', 'Pclass']]
tickets.reset_index(drop=True, inplace=True)
tickets

Name            Ticket  \
0                              Braund, Mr. Owen Harris         A/5 21171   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...          PC 17599   
2                               Heikkinen, Miss. Laina  STON/O2. 3101282   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)            113803   
4                             Allen, Mr. William Henry            373450   
..                                                 ...               ...   
886                              Montvila, Rev. Juozas            211536   
887                       Graham, Miss. Margaret Edith            112053   
888           Johnston, Miss. Catherine Helen "Carrie"        W./C. 6607   
889                              Behr, Mr. Karl Howell            111369   
890                                Dooley, Mr. Patrick            370376   

        Fare Cabin  Pclass  
0     7.2500   NaN       3  
1    71.2833   C85       1  
2     7.9250   NaN       3  
3    53.1000  C123       1  
4     8.0500   NaN       3  
..       ...   ...     ...  
886  13.0000   NaN       2  
887  30.0000   B42       1  
888  23.4500   NaN       3  
889  30.0000  C148       1  
890   7.7500   NaN       3  

[891 rows x 5 columns]

In [192]:
print(len(tickets['Ticket'].unique().tolist()))

681


Estrazione statistiche targets:

In [193]:
targets = tickets[['Pclass', 'Fare']].groupby(by='Pclass', as_index=False).mean()
targets['Fare'] = targets['Fare'].round(4)
targets

Pclass     Fare
0       1  84.1547
1       2  20.6622
2       3  13.6756

In [194]:
fares_map = {i: targets['Fare'].loc[i-1] for i in targets['Pclass']}
fares_map

{1: 84.1547, 2: 20.6622, 3: 13.6756}

Aggiunta persone esterne:

In [195]:
real_extras = ['Hershey, Mr. Milton',
                'Morgan, Mr. J. Pierpont',
                'Marconi, Mr. Guglielmo',
                'Frick, Mr. Henry Clay',
                'Vanderbilt, Mr. Alfred Gwynne',
                'Dreiser, Mr. Theodore',
                'Mott, Mr. John',
                'Penney, Mr. James Cash',
                'Seiberling, Mr. Frank',
                'Dix, Mr. John Alden',
                'Adams, Mr. Henry']

In [196]:
fakes = pd.read_csv("../Dataset/Titanic/nomi_fake.csv", sep=';')
fakes.head()

Surname, Name
0        Sato, Mr. Hakaru
1  Takahashi, Mr. Haruaki
2    Tanaka, Mr. Haruhisa
3  Watanabe, Mr. Haruichi
4         Ito, Mr. Haruki

Add extras:

In [197]:
import random
ls_extras = real_extras + fakes['Surname, Name'].tolist()
ls_names = []
ls_tickets = []
ls_pclasses = []
ls_fares = []

for extra in ls_extras:
    ls_names.append(extra)
    ls_tickets.append(f"ET {random.randint(10000000, 99999999)}")
    pclass = 1 if extra in real_extras else random.randint(1, 3)
    ls_pclasses.append(pclass)
    ls_fares.append(fares_map[pclass])

extras = pd.DataFrame({'Name': ls_names, 'Ticket': ls_tickets, 'Fare': ls_fares, 'Pclass': ls_pclasses})
extras.head(15)


Name       Ticket     Fare  Pclass
0             Hershey, Mr. Milton  ET 98427859  84.1547       1
1         Morgan, Mr. J. Pierpont  ET 10013244  84.1547       1
2          Marconi, Mr. Guglielmo  ET 49193462  84.1547       1
3           Frick, Mr. Henry Clay  ET 80532633  84.1547       1
4   Vanderbilt, Mr. Alfred Gwynne  ET 71915108  84.1547       1
5           Dreiser, Mr. Theodore  ET 48830724  84.1547       1
6                  Mott, Mr. John  ET 73428580  84.1547       1
7          Penney, Mr. James Cash  ET 65613343  84.1547       1
8           Seiberling, Mr. Frank  ET 76226659  84.1547       1
9             Dix, Mr. John Alden  ET 15168525  84.1547       1
10               Adams, Mr. Henry  ET 28340788  84.1547       1
11               Sato, Mr. Hakaru  ET 54910034  84.1547       1
12         Takahashi, Mr. Haruaki  ET 34942826  13.6756       3
13           Tanaka, Mr. Haruhisa  ET 21021030  13.6756       3
14         Watanabe, Mr. Haruichi  ET 79331706  13.6756       3

In [198]:
tickets = pd.concat([tickets, extras])
tickets

Name            Ticket  \
0                             Braund, Mr. Owen Harris         A/5 21171   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...          PC 17599   
2                              Heikkinen, Miss. Laina  STON/O2. 3101282   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)            113803   
4                            Allen, Mr. William Henry            373450   
..                                                ...               ...   
50                              Sakamoto, Mrs. Konomi       ET 93720440   
51                              Murakami, Mrs. Koruri       ET 75553455   
52                                    Ota, Mrs. Kotoe       ET 49568397   
53                                   Kaneko, Mrs. Kou       ET 51027651   
54                                  Fujii, Mrs. Kozue       ET 91381109   

       Fare Cabin  Pclass  
0    7.2500   NaN       3  
1   71.2833   C85       1  
2    7.9250   NaN       3  
3   53.1000  C123       1  
4    8.0500   NaN       3  
..      ...   ...     ...  
50  20.6622   NaN       2  
51  13.6756   NaN       3  
52  20.6622   NaN       2  
53  20.6622   NaN       2  
54  20.6622   NaN       2  

[946 rows x 5 columns]

Prepare for export:

In [199]:
tickets.sort_values(by='Ticket', inplace=True)
tickets['Ticket Class'] = [class_names[tickets['Pclass'].iloc[i]] for i in range(len(tickets.index))]
tickets.rename(columns={'Name': 'Name Orig'}, inplace=True)
tickets.reset_index(drop=True, inplace=True)
tickets

Name Orig       Ticket    Fare  \
0                                 Cherry, Miss. Gladys       110152  86.500   
1    Rothes, the Countess. of (Lucy Noel Martha Dye...       110152  86.500   
2                                Maioni, Miss. Roberta       110152  86.500   
3               Taussig, Mrs. Emil (Tillie Mandelbaum)       110413  79.650   
4                                  Taussig, Miss. Ruth       110413  79.650   
..                                                 ...          ...     ...   
941                       Harknett, Miss. Alice Phoebe   W./C. 6609   7.550   
942                        Chaffee, Mr. Herbert Fuller  W.E.P. 5734  61.175   
943                                 Harris, Mr. Walter    W/C 14208  10.500   
944                            Crosby, Miss. Harriet R    WE/P 5735  71.000   
945                       Crosby, Capt. Edward Gifford    WE/P 5735  71.000   

    Cabin  Pclass  Ticket Class  
0     B77       1   First Class  
1     B77       1   First Class  
2     B79       1   First Class  
3     E67       1   First Class  
4     E68       1   First Class  
..    ...     ...           ...  
941   NaN       3   Third Class  
942   E31       1   First Class  
943   NaN       2  Second Class  
944   B22       1   First Class  
945   B22       1   First Class  

[946 rows x 6 columns]

In [200]:
# tickets['Surname'] = tickets['Name Orig'].apply(lambda x: x.split(", ")[0])
# tickets['Name'] = tickets['Name Orig'].apply(lambda x: x.split(", ")[1])
# tickets_ok = pd.DataFrame({'Surname': tickets['Surname'], 'Name': tickets['Name'], 'Ticket': tickets['Ticket'], 'Fare': tickets['Fare'], 'Cabin': tickets['Cabin'], 'Ticket Class': tickets['Ticket Class']})
# tickets_ok

In [201]:
#tickets_ok.to_csv("../out/Titanic/tickets.csv", index=False)

JSON version:

In [202]:
from datetime import datetime, timedelta
import json, random

sale_start = datetime.strptime("1911-05-31 12:00:00", "%Y-%m-%d %H:%M:%S")
sale_end = datetime.strptime("1912-04-10 11:00:00", "%Y-%m-%d %H:%M:%S")
delta_minutes = int((sale_end - sale_start).total_seconds() / 60)
print(delta_minutes)
print(sale_start + timedelta(minutes=delta_minutes))

453540
1912-04-10 11:00:00


In [203]:
tickets_sold = []
warned = []
bls = ['No', 'Yes']
#pets = ['Dog', 'Cat', 'Bird']

for t in tickets['Ticket'].unique().tolist():
    filt = tickets.loc[tickets['Ticket'] == t].reset_index()
    buyer = filt['Name Orig'].iloc[0]
    purchase_time = datetime.strftime(sale_start + timedelta(minutes=random.randint(0,delta_minutes)),"%Y-%m-%d %H:%M:%S")
    #print(purchase_time)
    for i in filt.index:
        buf = {}
        buf['Ticket'] = t
        buf['Purchase Time'] = purchase_time
        buf['Buyer Name'] = buyer
        buf['Holder Name'] = filt['Name Orig'].iloc[i]
        buf['Fare'] = filt['Fare'].iloc[i]
        buf['Ticket Class'] = filt['Ticket Class'].iloc[i]
        buf['Cabin'] = filt['Cabin'].iloc[i]
        has_pet = random.randint(0, 1)
        buf['Has Pet'] = bls[has_pet]
        tickets_sold.append(buf)

json.dump(tickets_sold, open("../out/Titanic/tickets_sold.json", 'w'))

#### **4) Famiglie**

In [204]:
families = full[['Name', 'SibSp', 'Parch']].loc[(full['SibSp'] > 0) | (full['Parch'] > 0)]
families.sort_values(by='Name', inplace=True)
families.rename(columns={'Name': 'Name Orig', 'SibSp': 'Sib Sp', 'Parch': 'Par Ch'}, inplace=True)
families['Surname'] = families['Name Orig'].apply(lambda x: x.split(", ")[0])
families['Name'] = families['Name Orig'].apply(lambda x: x.split(", ")[1])
families.reset_index(drop=True, inplace=True)
families

Name Orig  Sib Sp  Par Ch  \
0                       Abbott, Mr. Rossmore Edward       1       1   
1                  Abbott, Mrs. Stanton (Rosa Hunt)       1       1   
2                               Abelson, Mr. Samuel       1       0   
3             Abelson, Mrs. Samuel (Hannah Wizosky)       1       0   
4    Ahlin, Mrs. Johan (Johanna Persdotter Larsson)       1       0   
..                                              ...     ...     ...   
349                            Zabour, Miss. Hileni       1       0   
350                           Zabour, Miss. Thamine       1       0   
351    de Messemaeker, Mrs. Guillaume Joseph (Emma)       1       0   
352                       del Carlo, Mr. Sebastiano       1       0   
353                 van Billiard, Mr. Austin Blyler       0       2   

            Surname                                     Name  
0            Abbott                      Mr. Rossmore Edward  
1            Abbott                 Mrs. Stanton (Rosa Hunt)  
2           Abelson                               Mr. Samuel  
3           Abelson             Mrs. Samuel (Hannah Wizosky)  
4             Ahlin  Mrs. Johan (Johanna Persdotter Larsson)  
..              ...                                      ...  
349          Zabour                             Miss. Hileni  
350          Zabour                            Miss. Thamine  
351  de Messemaeker             Mrs. Guillaume Joseph (Emma)  
352       del Carlo                           Mr. Sebastiano  
353    van Billiard                        Mr. Austin Blyler  

[354 rows x 5 columns]

In [205]:
print(len(families['Surname'].unique().tolist()))

190


In [206]:
families_ok = families[['Surname', 'Name', 'Sib Sp', 'Par Ch']]
families_ok

Surname                                     Name  Sib Sp  Par Ch
0            Abbott                      Mr. Rossmore Edward       1       1
1            Abbott                 Mrs. Stanton (Rosa Hunt)       1       1
2           Abelson                               Mr. Samuel       1       0
3           Abelson             Mrs. Samuel (Hannah Wizosky)       1       0
4             Ahlin  Mrs. Johan (Johanna Persdotter Larsson)       1       0
..              ...                                      ...     ...     ...
349          Zabour                             Miss. Hileni       1       0
350          Zabour                            Miss. Thamine       1       0
351  de Messemaeker             Mrs. Guillaume Joseph (Emma)       1       0
352       del Carlo                           Mr. Sebastiano       1       0
353    van Billiard                        Mr. Austin Blyler       0       2

[354 rows x 4 columns]

In [207]:
families_ok.to_csv("../out/Titanic/families.csv", index=False)

#### **5) Sopravvissuti**

In [208]:
survived = full[['Name', 'Sex', 'Age']].loc[full['Survived'] == 1]
survived.rename(columns={'Name': 'Name Orig'}, inplace=True)
survived['Surname'] = survived['Name Orig'].apply(lambda x: x.split(", ")[0])
survived['Name'] = survived['Name Orig'].apply(lambda x: x.split(", ")[1])
survived.reset_index(drop=True, inplace=True)
survived

Name Orig     Sex   Age  \
0    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
1                               Heikkinen, Miss. Laina  female  26.0   
2         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
3    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0   
4                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0   
..                                                 ...     ...   ...   
337                   Najib, Miss. Adele Kiamie "Jane"  female  15.0   
338      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0   
339       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0   
340                       Graham, Miss. Margaret Edith  female  19.0   
341                              Behr, Mr. Karl Howell    male  26.0   

       Surname                                        Name  
0      Cumings  Mrs. John Bradley (Florence Briggs Thayer)  
1    Heikkinen                                 Miss. Laina  
2     Futrelle          Mrs. Jacques Heath (Lily May Peel)  
3      Johnson    Mrs. Oscar W (Elisabeth Vilhelmina Berg)  
4       Nasser                 Mrs. Nicholas (Adele Achem)  
..         ...                                         ...  
337      Najib                   Miss. Adele Kiamie "Jane"  
338     Potter       Mrs. Thomas Jr (Lily Alexenia Wilson)  
339    Shelley         Mrs. William (Imanita Parrish Hall)  
340     Graham                        Miss. Margaret Edith  
341       Behr                             Mr. Karl Howell  

[342 rows x 5 columns]

In [209]:
survived_ok = survived[['Surname', 'Name', 'Sex', 'Age']]
survived_ok['Rescue Ship'] = ['Carpathia' for i in survived_ok.index]
survived_ok['Arrival Pier'] = ['Pier 54' for i in survived_ok.index]
survived_ok['Arrival Harbour'] = ['New York' for i in survived_ok.index]
survived_ok['Arrival Time'] = [datetime.strptime("1912-04-18 21:30:00", "%Y-%m-%d %H:%M:%S") + timedelta(seconds=random.randint(0,342)) for i in survived_ok.index]
survived_ok

Surname                                        Name     Sex   Age  \
0      Cumings  Mrs. John Bradley (Florence Briggs Thayer)  female  38.0   
1    Heikkinen                                 Miss. Laina  female  26.0   
2     Futrelle          Mrs. Jacques Heath (Lily May Peel)  female  35.0   
3      Johnson    Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0   
4       Nasser                 Mrs. Nicholas (Adele Achem)  female  14.0   
..         ...                                         ...     ...   ...   
337      Najib                   Miss. Adele Kiamie "Jane"  female  15.0   
338     Potter       Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0   
339    Shelley         Mrs. William (Imanita Parrish Hall)  female  25.0   
340     Graham                        Miss. Margaret Edith  female  19.0   
341       Behr                             Mr. Karl Howell    male  26.0   

    Rescue Ship Arrival Pier Arrival Harbour        Arrival Time  
0     Carpathia      Pier 54        New York 1912-04-18 21:33:41  
1     Carpathia      Pier 54        New York 1912-04-18 21:32:10  
2     Carpathia      Pier 54        New York 1912-04-18 21:33:44  
3     Carpathia      Pier 54        New York 1912-04-18 21:35:27  
4     Carpathia      Pier 54        New York 1912-04-18 21:35:10  
..          ...          ...             ...                 ...  
337   Carpathia      Pier 54        New York 1912-04-18 21:31:56  
338   Carpathia      Pier 54        New York 1912-04-18 21:31:37  
339   Carpathia      Pier 54        New York 1912-04-18 21:33:03  
340   Carpathia      Pier 54        New York 1912-04-18 21:34:55  
341   Carpathia      Pier 54        New York 1912-04-18 21:33:50  

[342 rows x 8 columns]

In [210]:
survived_ok.to_csv("../out/Titanic/arrival_registry.csv", index=False)

TRAIN / TEST SPLIT:
* Train set: IDs da 1 a 891
* Test set: IDs da 892 a 1309 (not utilized here)
* Features: pclass, sex, age, sib_sp, par_ch, fare

In [211]:
#gendsub['Survived'] = np.where(gendsub['Survived Bool']==1, 'Yes', 'No')
#gendsub.to_csv("../out/Titanic/casualties_full.csv", index=False)